csi/moac/volume.ts

```diff
@@ -143,57 +141,78 @@ export class Volume {
         'Cannot publish a volume that is neither healthy nor degraded'
       );
     }
-    let nexus = this.nexus;
-    if (!nexus) {
-      // Ensure replicas can be accessed from nexus. Set share protocols.
-      const [nexusNode, replicaSet] = await this._ensureReplicaShareProtocols();
-      nexus = await this._createNexus(nexusNode, replicaSet);
-    } else {
-      log.debug(`Publishing volume ${this} that already has a nexus`)
-    }
-    let uri = nexus.getUri();
-    if (!uri) {
-      uri = await nexus.publish(protocol);
-    } else {
-      log.debug(`Publishing volume ${this} that has been already published`)
+
+    // FSA will take care of publishing the nexus
+    this.publishedOn = this._desiredNexusNode(this._activeReplicas()).name;
+    this.fsa();
```

Comment:
How soon will the real fsa run after calling this fn?

Reply:
just an eye blink - in next reactor loop iteration of nodejs.